src/Common/ProxyConfigurationResolverProvider.cpp

```diff
@@ -227,7 +234,7 @@ std::shared_ptr<ProxyConfigurationResolver> ProxyConfigurationResolverProvider::
      * In case the combination of config_prefix and configuration does not provide a resolver, try to get it from general / new settings.
      * Falls back to Environment resolver if no configuration is found.
      * */
-    return ProxyConfigurationResolverProvider::get(Protocol::ANY, configuration);
+    return ProxyConfigurationResolverProvider::get(request_protocol, configuration);
```

Comment:
This fixes a bug. We were supposed to be flexible only on the old settings format, but when fallback to new settings, it shouldn't be flexible.